File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int i=digits.size()-1;
        if(digits[i]==9){
            int count=0;
            while(i>=0 && digits[i]==9){
                count++;
                digits.pop_back();
                i--;
            }
            if(i>=0){
                digits[i]++;
            }
            else{
                digits.push_back(1);
            }
            for(int i=0;i<count;i++){
                digits.push_back(0);
            }
            return digits;
        }
        digits[i]++;
        return digits;
    }
};

File: 0066-plus-one/0066-plus-one.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        int i=digits.Length-1;
        if(digits[i]==9){    
            int count=0;
            while(i>=0 && digits[i]==9){
                count++;
                i--;
            }
            int[] temp;
            if(i<0){
                temp= new int[digits.Length+1];
                i++;
                temp[i]=1;
            }
            else{
                temp= new int[digits.Length];
                Array.Copy(digits,temp,i+1);
                temp[i]++;
            }
            i++;
            for(int j=0;j<count;j++){
                temp[i]=0;
                i++;
            }
            return temp;
        }
        digits[i]++;
        return digits;
    }
}